src/Lean/Elab/BuiltinTerm.lean

```diff
@@ -190,7 +190,8 @@ private def mkFreshTypeMVarFor (expectedType? : Option Expr) : TermElabM Expr :=
     | some val => pure val
     | none     => throwIllFormedSyntax
   let typeMVar ← mkFreshTypeMVarFor expectedType?
-  let u ← getDecLevel typeMVar
+  let some u ← getDecLevel? typeMVar
+    | throwError s!"Cannot elab numeral. Universe level of expected type must be greater than 0."
```

Comment:
The error message of `getDecLevel?` includes the universe level, and I think we shouldn't lose that:
```suggestion
  let u' ← getLevel typeMVar
  let some u ← decLevel? u'
    | throwError s!"Universe level of expected type must be greater than 0, but it is {u'}."
```
Alternatively:
```suggestion
  let some u ← getDecLevel? typeMVar
    | throwError s!"Expected type is{indentD (← inferType typeMVar)}\nbut universe level of expected type must be greater than 0."
```

Reply:
Thanks for the suggestion @kmill ! I particularly like the second approach, which I extended a bit. What do you think:

```
Tried to elab numeric literal
  1
of type 
  Nat → True : Prop
but universe level of
  Prop
is not greater than 0.
```

I think the piece of information I was missing here previously is that the error is about the "numeric literal" `1`.